common/src/main/java/io/netty/util/concurrent/Future.java

```diff
@@ -341,8 +303,47 @@ default V get(long timeout, TimeUnit unit) throws InterruptedException, Executio
     }
 
     /**
-     * Returns a {@link FutureCompletionStage} that reflects the state of this {@link Future} and so will receive
-     * all updates as well.
+     * Create a new {@link Future} that will complete with the result of this {@link Future} mapped through the given
+     * mapper function.
+     * <p>
+     * If this future fails, then the returned future will fail as well, with the same exception. Cancellation of either
+     * future will cancel the other. If the mapper function throws, the returned future will fail, but this future will
+     * be unaffected.
+     *
+     * @param mapper The function that will convert the result of this future into the result of the returned future.
+     * @param <R>    The result type of the mapper function, and of the returned future.
+     * @return A new future instance that will complete with the mapped result of this future.
+     */
+    default <R> Future<R> map(Function<V, R> mapper) {
+        return Futures.map(this, mapper);
+    }
+
+    /**
+     * Create a new {@link Future} that will complete with the result of this {@link Future} flat-mapped through the
+     * given mapper function.
+     * <p>
+     * The "flat" in "flat-map" means the given mapper function produces a result that itself is a future-of-R, yet this
```

Comment:
In ServiceTalk we found that providing an imperative code equivalent was useful to understand more than describing in words. As an example see:

https://apple.github.io/servicetalk//servicetalk/SNAPSHOT/javadoc/io/servicetalk/concurrent/api/Single.html#flatMap(java.util.function.Function)